doc/specs/#2871 - Pane Navigation/#2871 - Pane Navigation.md

```diff
@@ -0,0 +1,298 @@
+
+// Focus the prev inOrder pane
+// - this seems straightforward
+{ "command": { "action": "focusPrevPane", "order": "inOrder" } },
+
+// Focus the next pane, in mru order, explicitly disable the switcher
+// - The user opted in to only being able to MRU switch one deep. That's fine, that's what they want.
+{ "command": { "action": "focusNextPane", "order": "mru", "useSwitcher": false} },
+
+// Focus the prev inOrder pane, explicitly with the switcher
+// - Maybe they disabled the switcher globally, but what it on for this action?
+{ "command": { "action": "focusPrevPane", "order": "inOrder", "useSwitcher": true } },
+```
+_From [discussion in the implementation
+PR](https://github.com/microsoft/terminal/pull/8183#issuecomment-729672645)_
+
+Boiled down, that's three actions:
+* `focusPane(target=id)`
+* `focusNextPane(order="inOrder|mru", useSwitcher=true|false)`
+* `focusPrevPane(order="inOrder|mru", useSwitcher=true|false)`
+
+* **Pros**:
+  - Everything is explicit, including the option to use the pane switcher (when
+    available)
+  - Adds support for in-order pane switching
+  - No "conditional parameters" - where providing one argument makes other
+    arguments invalid or ambiguous.
+* **Cons**:
+  - Doesn't really address any of the other scenarios
+  - What does the "next most-recently-used tab" even mean? How is it different
+    than "previous most-recently-used tab"? Semantically, these are the same
+    thing!
+  - No one's even asked for in-order pane switching. Is that a UX that even
+    really makes sense?
+
+❌ This proposal is no longer being considered.
+
+> 👉 **NOTE**: At this point, we stopped considering navigating in both MRU
+> "directions", since both the next and prev MRU pane are the same thing. We're
+> now using "last" to mean "the previous MRU pane".
+
+### Proposal C: One actions, combine the args
+
+* `moveFocus(target=id|"up|down|left|right|last")`
+
+* **Pros**:
+  - Absolutely the least complicated action to author. There's only one
+    parameter, `target`.
+  - No "conditional parameters".
+* **Cons**:
+  - How do we express this in the Settings UI? Mixed-type enums work fine for
+    the font weight, where each enum value has a distinct integer value it maps
+    to, but in this case, using `id` is entirely different from the other
+    directional values
+
+❌ This proposal is no longer being considered.
+
+### Proposal D: Two actions
+
+* `focusPane(target=id)`
+* `moveFocus(direction="up|down|left|right|last")`
+
+* **Pros**:
+  - Each action does explicitly one thing.
+* **Cons**:
+  - two actions for _similar_ behavior
+  - This now forks the "Direction" enum into "MoveFocusDirection" and
+    "ResizeDirection" (because `resizePane(last)` doesn't make any sense).
+
+This proposal doesn't really have any special consideration for the pane
+switcher UX. Neither of these actions would summon the pane switcher UX.
+
+### Proposal E: Three actions

```

Comment:
Should Proposal E and F also have "❌ This proposal is no longer being considered."?